### PR TITLE
mergify: Remove approved review when a PR changes

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

When people have reviewed a PR, and the contributor updates it, the
reviews are staying valid. This is unfortunate as the PR might need new
corrections.

Ideally a positive review (Approved) needs to be given again. And there
needs to be a confirmation that a negative review (Changes requested)
has been addressed.

## Related issues ##

Closes: #505

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -10,5 +10,7 @@ pull_request_rules:
         method: rebase
         rebase_fallback: merge
         strict: smart
-      dismiss_reviews: {}
+      dismiss_reviews:
```
Comment (ShyamsundarR) [inline .mergify.yml]: Should this be a separate rule name and set of actions. I somehow cannot understand a `condition` that states check for > 2 approving issues having an action to remove stale reviews. IOW, if the conditions are not met the actions won't be performed, whereas we want to remove any approving reviews if a PR is updated (not only when it is ready for merge).

**Comment by nixpanic** [comment]
On Tue, Sep 03, 2019 at 04:30:46AM -0700, ShyamsundarR wrote:
> Should this be a separate rule name and set of actions. I somehow
> cannot understand a `condition` that states check for > 2 approving
> issues having an action to remove stale reviews. IOW, if the
> conditions are not met the actions won't be performed, whereas we want
> to remove any approving reviews if a PR is updated (not only when it
> is ready for merge).

Hmm, maybe. I'll read the docs again and see if a 2nd rule makes more
sense.

File: .mergify.yml (hunk 1 above)
Comment (nixpanic) [reply .mergify.yml]: A 2nd rule has been added

**Comment by Madhu-1** [comment]
@nixpanic as @ShyamsundarR approved this PR. can you  modify something for this  PR, let's see what happens

**Comment by nixpanic** [comment]
On Tue, Sep 03, 2019 at 09:27:47PM -0700, Madhu Rajanna wrote:
> @nixpanic as @ShyamsundarR approved this PR. can you  modify something
> for this  PR, let's see what happens

Lets see?

**Comment by Madhu-1** [comment]
@nixpanic approval is not removed :(

**Comment by nixpanic** [comment]
On Wed, Sep 04, 2019 at 01:58:55AM -0700, Madhu Rajanna wrote:
> @nixpanic approval is not removed :(

No, but that can well be caused by the way mergify works. Maybe it needs
to be merged before it applies the new rules.

I've dropped the 2nd [DNM] commit again to make this PR in mergable
shape.

**Comment by Madhu-1** [comment]
approved, can you  open a test PR  to test this works as expected

**Comment by Madhu-1** [comment]
>The new Mergify configuration is valid
This pull request must be merged manually because it modifies Mergify configuration

merging this  one manually as per mergify

**Comment by nixpanic** [comment]
On Wed, Sep 04, 2019 at 02:16:54AM -0700, Madhu Rajanna wrote:
> approved, can you  open a test PR  to test this works as expected

See PR #587

**Pull request merged**